src/library/views.py

```diff
@@ -54,9 +55,22 @@ def campaigns_library_list(request):
     """
 
     with library_schema_context():
-        # Get the active quests and force the query to run while still in the library schema
-        # by calling list() on the queryset
-        library_categories = list(Category.objects.filter(active=True))
+        # Filter quests that are visible to students and not archived,
+        # linked to the campaign (category) via OuterRef for subquery use.
+        filtered_quests = Quest.objects.filter(
```

Comment:
Let's put this logic in a method in the Category's Manager class (which it doesn't have at the moment, but you can see almost every other model does have one).

Then we can just do `library_categories = Category.objects.all_active_with_available_quests()` in the view.

And that also allows us to properly unit test the `all_active_with_available_quests()` separately from the view method.

I'm not sure this is the cleanest way to do the query, but lets refactor into a Category manager method first, with tests, then we can tinker with the insides of the method if needed.

Reply:
Some reading for you:  https://docs.djangoproject.com/en/4.2/topics/db/managers/

Reply:
Can you explain that more?  When I first made most of the models I went overkill with creating model QuerySet classes.  I think because this is the only one we have, it's fine to access via the a manager...unless you have a good reason to add both Manager and QuerySet for this single method?